Refactor GroupChatForm

noavarice opened the pull request:
Change public method to be a private slot, as it was initially intended + some other small changes

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/4573)
<!-- Reviewable:end -->

noavarice committed:
refactor: GroupChatForm::onUserListChanged is a slot now

noavarice committed:
refactor: 'correctNames' function refactoring

'correctNames' function moved from class, renamed and refactored
according to its purpose

noavarice committed:
refactor: onUserListChanged refactored

noavarice committed:
refactor: headers order is changed properly

Diadlo commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[src/widget/form/groupchatform.cpp, line 50 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR1Mlxuprxydtu1rkK:-KrR1Mlxuprxydtu1rkL:bmqsuq5) ([raw file](https://github.com/qtox/qtox/blob/29d921a74c22826ba0480b85d43b4f6321c4bc40/src/widget/form/groupchatform.cpp#L50)):*
> ```cpp
> QString editName(const QString& name)
> {
>     const int pos = name.indexOf(QRegExp("\n|\r\n|\r"));
> ```

I think `\r\n` here is overhead. Because `\r`position will be the same

---

*[src/widget/form/groupchatform.cpp, line 58 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR4Fc1IEJ4k9SoBC12:-KrR4Fc2WQ159VK6K2dH:b-4lsmgk) ([raw file](https://github.com/qtox/qtox/blob/29d921a74c22826ba0480b85d43b4f6321c4bc40/src/widget/form/groupchatform.cpp#L58)):*
> ```cpp
>     const int len = result.length();
>     result.chop(len - pos);
>     result.append("...");
> ```

Ellipsis has its own symbol. Better use it: `…`

---

*[src/widget/form/groupchatform.cpp, line 103 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR3t-sIJQOcxDtrPew:-KrR3t-sIJQOcxDtrPex:b7h5mqd) ([raw file](https://github.com/qtox/qtox/blob/29d921a74c22826ba0480b85d43b4f6321c4bc40/src/widget/form/groupchatform.cpp#L103)):*
> ```cpp
>     const QStringList names(group->getPeerList());
>     for (const QString& fullName : names) {
>         const QString editedName = editName(fullName);
> ```

Why here is name without coma?

---

*[src/widget/form/groupchatform.cpp, line 104 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR1py67OvdvUkSyQkz:-KrR1py67OvdvUkSyQl-:bu88k4b) ([raw file](https://github.com/qtox/qtox/blob/29d921a74c22826ba0480b85d43b4f6321c4bc40/src/widget/form/groupchatform.cpp#L104)):*
> ```cpp
>     for (const QString& fullName : names) {
>         const QString editedName = editName(fullName);
>         QLabel* l = new QLabel(editedName);
> ```

Why not use 1 label?
Where is `delete` for those labels?

---

*[src/widget/form/groupchatform.cpp, line 173 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR3d0Mo8BNgQVaAiwB:-KrR3d0Mo8BNgQVaAiwC:b-ynmr81) ([raw file](https://github.com/qtox/qtox/blob/29d921a74c22826ba0480b85d43b4f6321c4bc40/src/widget/form/groupchatform.cpp#L173)):*
> ```cpp
> /**
>  * @brief This slot is intended to connect to Group::userListChanged signal.
>  * Brief list of actions made by slot:
> ```

List of actions in function description => maybe something goes wrong. What about split on 3 functions?

---

*[src/widget/form/groupchatform.cpp, line 223 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR2jBwG7pvorVJsmsa:-KrR2jBwG7pvorVJsmsb:bwxbvad) ([raw file](https://github.com/qtox/qtox/blob/29d921a74c22826ba0480b85d43b4f6321c4bc40/src/widget/form/groupchatform.cpp#L223)):*
> ```cpp
>     }
> 
>     auto comparator = [](const QLabel* a, const QLabel* b)
> ```

Why extracted? If you give name for a lambda, probably, it shouldn't be the lambda, but function :)
(Lambda function is anonymous function)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[src/widget/form/groupchatform.cpp, line 189 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrRGZbc3RA5dlWlFxF2:-KrRGZbc3RA5dlWlFxF3:b-imlwqy) ([raw file](https://github.com/qtox/qtox/blob/29d921a74c22826ba0480b85d43b4f6321c4bc40/src/widget/form/groupchatform.cpp#L189)):*
> ```cpp
> 
>     QLayoutItem* child;
>     while (child = namesListLayout->takeAt(0)) {
> ```

Please, revert this change. Extra braces used to silence warning about assignment in condition

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

noavarice commented:
Review status: 2 of 3 files reviewed at latest revision, 7 unresolved discussions.

---

*[src/widget/form/groupchatform.cpp, line 50 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR1Mlxuprxydtu1rkK:-KrXqEMD5wewjSnUTSnV:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L50)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

I think `\r\n` here is overhead. Because `\r`position will be the same
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 58 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR4Fc1IEJ4k9SoBC12:-KrXqDbJ1nn9g8v5gM9d:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L58)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Ellipsis has its own symbol. Better use it: `…`
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 103 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR3t-sIJQOcxDtrPew:-KrXpvWMaUP8CxkSb36d:b-7795wk) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L103)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Why here is name without coma?
</blockquote></details>

This is a bug. Since `GroupChatForm` is creating from empty group, there is no members' names at all, so I removed this code part

---

*[src/widget/form/groupchatform.cpp, line 104 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR1py67OvdvUkSyQkz:-KrXrxXPJ8qZKKPrqRvo:b-8lsl6m) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L104)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Why not use 1 label?
Where is `delete` for those labels?
</blockquote></details>

1) There's style applying several times for concrete user name, it's much easier with single `QLabel` object than using string magic
2) `namesListLayout` must be always responsible for `QLabel`s' destruction

---

*[src/widget/form/groupchatform.cpp, line 173 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR3d0Mo8BNgQVaAiwB:-KrXt-cXhJsjL2J19JgM:b-emv0v) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L173)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

List of actions in function description => maybe something goes wrong. What about split on 3 functions?
</blockquote></details>

It's single logic action -- form updating according to user list changes. Also there is no code duplication to split it on several other functions. I am just not sure about it

---

*[src/widget/form/groupchatform.cpp, line 189 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrRGZbc3RA5dlWlFxF2:-KrXppMHScAObKGMoLvT:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L189)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Please, revert this change. Extra braces used to silence warning about assignment in condition
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 223 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR2jBwG7pvorVJsmsa:-KrXpKBvZ5wrxvuOL9Tu:b-1wl3j3) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L223)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Why extracted? If you give name for a lambda, probably, it shouldn't be the lambda, but function :)
(Lambda function is anonymous function)
</blockquote></details>

Only for code readability. I see no reason to make it a separate function, but if you really want, I can revert this change too

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[src/widget/form/groupchatform.cpp, line 58 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR4Fc1IEJ4k9SoBC12:-KrZPjLGlizn7wiNKVAK:bxrek0z) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L58)):*
<details><summary><i>Previously, noavarice wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Hm... Reviewable show here `\u2026`. Do you wrote it? Or just reviewable bug?
If it's you, why not use `…` symbol directly?

---

*[src/widget/form/groupchatform.cpp, line 103 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR3t-sIJQOcxDtrPew:-KrZQ9UIdHt99DluWwbh:b-oh8d5y) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L103)):*
<details><summary><i>Previously, noavarice wrote…</i></summary><blockquote>

This is a bug. Since `GroupChatForm` is creating from empty group, there is no members' names at all, so I removed this code part
</blockquote></details>

Group shouldn't be empty. Here should be 1 member -- you. Probably, it's a [bug](https://github.com/qTox/qTox/issues/4503). Prabably, should be added createing 1 label? Or something like that?

---

*[src/widget/form/groupchatform.cpp, line 104 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR1py67OvdvUkSyQkz:-KrZQbyJgZZm_UAAcSwa:b-4v0lum) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L104)):*
<details><summary><i>Previously, noavarice wrote…</i></summary><blockquote>

1) There's style applying several times for concrete user name, it's much easier with single `QLabel` object than using string magic
2) `namesListLayout` must be always responsible for `QLabel`s' destruction
</blockquote></details>

1. I thought you care about memory ;)
2. I'm not sure, but now it doesn't matter

---

*[src/widget/form/groupchatform.cpp, line 173 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR3d0Mo8BNgQVaAiwB:-KrZRL0QSk1m2O6Q1JVp:b-a4yzvj) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L173)):*
<details><summary><i>Previously, noavarice wrote…</i></summary><blockquote>

It's single logic action -- form updating according to user list changes. Also there is no code duplication to split it on several other functions. I am just not sure about it
</blockquote></details>

1. There is code duplication: update peer count exists in `retanslateUi()`, creating label was in constructor (now removed)
2. Split on few functions used not only for reducing code duplication. But to make code easier to read. So you give a name to each action
```
updatePeerCount();
nickLabelList = loadPeerLabels();
fillNamesListLayut(nickLabelList); // probably, this is extra
updateCallButton();
```
Simple mnemonic rule: code of function should fit on one screen

---

*[src/widget/form/groupchatform.cpp, line 223 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR2jBwG7pvorVJsmsa:-KrZUTt4S-jf6mScf7M3:bru77uf) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L223)):*
<details><summary><i>Previously, noavarice wrote…</i></summary><blockquote>

Only for code readability. I see no reason to make it a separate function, but if you really want, I can revert this change too
</blockquote></details>

But you already make it a separate function. I just asking you to place it outside of this function or revert

---

*[src/widget/form/groupchatform.cpp, line 213 at r2](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrZUd9cdAEFUzys70JL:-KrZUd9cdAEFUzys70JM:b-4o3uqu) ([raw file](https://github.com/qtox/qtox/blob/0c703b7fca89d7b33d434af461c0a148b365eed8/src/widget/form/groupchatform.cpp#L213)):*
> ```cpp
>     auto comparator = [](const QLabel* a, const QLabel* b)
>     {
>         return a->text().toLower() < b->text().toLower();
> ```

Whe `.toLower()` is used here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

noavarice commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[src/widget/form/groupchatform.cpp, line 58 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR4Fc1IEJ4k9SoBC12:-KrZtcogsfZPBEJu8ugu:b-1oyog6) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L58)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Hm... Reviewable show here `\u2026`. Do you wrote it? Or just reviewable bug?
If it's you, why not use `…` symbol directly?
</blockquote></details>

I don't know, just found it as a Unicode symbol

---

*[src/widget/form/groupchatform.cpp, line 104 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR1py67OvdvUkSyQkz:-KrZu34qCgixXSidIkIy:br8xh2b) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L104)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

1. I thought you care about memory ;)
2. I'm not sure, but now it doesn't matter
</blockquote></details>

1) Yes, I do, but this is just out of PR)

---

*[src/widget/form/groupchatform.cpp, line 213 at r2](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrZUd9cdAEFUzys70JL:-KrZueInhQ1A-4wndZgU:biyfmqp) ([raw file](https://github.com/qtox/qtox/blob/0c703b7fca89d7b33d434af461c0a148b365eed8/src/widget/form/groupchatform.cpp#L213)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Whe `.toLower()` is used here?
</blockquote></details>

To sort names starting with same letter without being depended on its case

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

noavarice commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[src/widget/form/groupchatform.cpp, line 173 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR3d0Mo8BNgQVaAiwB:-Kr_KtsWza-cywffvVXd:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L173)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

1. There is code duplication: update peer count exists in `retanslateUi()`, creating label was in constructor (now removed)
2. Split on few functions used not only for reducing code duplication. But to make code easier to read. So you give a name to each action
```
updatePeerCount();
nickLabelList = loadPeerLabels();
fillNamesListLayut(nickLabelList); // probably, this is extra
updateCallButton();
```
Simple mnemonic rule: code of function should fit on one screen
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 223 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR2jBwG7pvorVJsmsa:-Kr_DhDfLAZ-ZHtz7zrA:bwvyox4) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L223)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

But you already make it a separate function. I just asking you to place it outside of this function or revert
</blockquote></details>

Reverted

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

noavarice commented:
Review status: 1 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/widget/form/groupchatform.cpp, line 58 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR4Fc1IEJ4k9SoBC12:-Krann-EiSgLXnSdSOGw:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L58)):*
<details><summary><i>Previously, noavarice wrote…</i></summary><blockquote>

I don't know, just found it as a Unicode symbol
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

---

Review status: 1 of 3 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573#-:-KrawKSpaTFvMEOVRQfC:b-jbsf8z)*
<!-- Sent from Reviewable.io -->

sudden6 commented:
Reviewed 1 of 3 files at r1, 1 of 2 files at r3.
Review status: 2 of 3 files reviewed at latest revision, 5 unresolved discussions.

---

*[src/widget/form/groupchatform.cpp, line 58 at r3](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrgHDEur6BFjT5ko2qG:-KrgHDEvTsQglAxSVfwA:b-unou2j) ([raw file](https://github.com/qtox/qtox/blob/8341173dde01bcb52bdc39eb6e6358cdf49bca7e/src/widget/form/groupchatform.cpp#L58)):*
> ```cpp
>     const int len = result.length();
>     result.chop(len - pos);
>     result.append("…");
> ```

leave a comment with the Unicode code + warning that it's not just three dots, else someone will change it and chaos ensues ;)

---

*[src/widget/form/groupchatform.cpp, line 121 at r3](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrgHoz8ZfM1W1w1gaR3:-KrgHoz9rXHQbA5WLCZG:bbowyqj) ([raw file](https://github.com/qtox/qtox/blob/8341173dde01bcb52bdc39eb6e6358cdf49bca7e/src/widget/form/groupchatform.cpp#L121)):*
> ```cpp
>     });
>     connect(group,
>             SIGNAL(userListChanged(uint32_t, const QMap<QByteArray, QString>&)),
> ```

don't use `SIGNAL` and `SLOT` macros, use a lambda function if needed

---

*[src/widget/form/groupchatform.cpp, line 162 at r3](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrgI4GOeOZNcHLYj8Mj:-KrgI4GOeOZNcHLYj8Mk:b1p97pg) ([raw file](https://github.com/qtox/qtox/blob/8341173dde01bcb52bdc39eb6e6358cdf49bca7e/src/widget/form/groupchatform.cpp#L162)):*
> ```cpp
>  * @brief This slot is intended to connect to Group::userListChanged signal.
>  * Brief list of actions made by slot:
>  *      1) sets text of how much people is in group;
> ```

`how many people are in the group`

---

*[src/widget/form/groupchatform.cpp, line 165 at r3](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrgIAqrGH51FHJT91MK:-KrgIAqsSZiJuSZizRjt:b-9p2qss) ([raw file](https://github.com/qtox/qtox/blob/8341173dde01bcb52bdc39eb6e6358cdf49bca7e/src/widget/form/groupchatform.cpp#L165)):*
> ```cpp
>  *      2) creates lexicographically sorted comma-separated list of user names, each name in its own
>  *      label;
>  *      3) sets call button style depending on peers count and etc.
> ```

`peer count`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

noavarice commented:
Review status: 2 of 3 files reviewed at latest revision, 5 unresolved discussions.

---

*[src/widget/form/groupchatform.cpp, line 103 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR3t-sIJQOcxDtrPew:-Krpw6YjuiOZ8YlnphjP:bfe1j1n) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L103)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Group shouldn't be empty. Here should be 1 member -- you. Probably, it's a [bug](https://github.com/qTox/qTox/issues/4503). Prabably, should be added createing 1 label? Or something like that?
</blockquote></details>

I think, this is out of this pull request since it's needed to change `Group`

---

*[src/widget/form/groupchatform.cpp, line 58 at r3](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrgHDEur6BFjT5ko2qG:-KrpxY-OCHlM7yUyOthD:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/8341173dde01bcb52bdc39eb6e6358cdf49bca7e/src/widget/form/groupchatform.cpp#L58)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

leave a comment with the Unicode code + warning that it's not just three dots, else someone will change it and chaos ensues ;)
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 121 at r3](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrgHoz8ZfM1W1w1gaR3:-Krpyu2wV8uckbjknekc:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/8341173dde01bcb52bdc39eb6e6358cdf49bca7e/src/widget/form/groupchatform.cpp#L121)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

don't use `SIGNAL` and `SLOT` macros, use a lambda function if needed
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 162 at r3](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrgI4GOeOZNcHLYj8Mj:-Krpx_NwppQromz5h7-b:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/8341173dde01bcb52bdc39eb6e6358cdf49bca7e/src/widget/form/groupchatform.cpp#L162)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

`how many people are in the group`
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 165 at r3](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrgIAqrGH51FHJT91MK:-Krpx_BcTsyDQ7YCYOjz:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/8341173dde01bcb52bdc39eb6e6358cdf49bca7e/src/widget/form/groupchatform.cpp#L165)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

`peer count`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

sudden6 commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573#-:-KrqP5ohsbiUtf9MZzgZ:b-jbsf8z)*
<!-- Sent from Reviewable.io -->

Talkless commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[src/widget/form/groupchatform.cpp, line 50 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR1Mlxuprxydtu1rkK:-Kruek-o7Un45hwrRF31:b-wvq6m3) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L50)):*
<details><summary><i>Previously, noavarice wrote…</i></summary><blockquote>

Done.
</blockquote></details>

It is recommended to use new QRegularExpression. See https://doc.qt.io/qt-5/qregularexpression.html#notes-for-qregexp-users .
Also, QStringLiteral("[...

---

*[src/widget/form/groupchatform.cpp, line 58 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-KruhAQHimtRiRy8vA0Q:-KruhAQHimtRiRy8vA0R:bxctj9u) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L58)):*
> ```cpp
>     const int len = result.length();
>     result.chop(len - pos);
>     result.append("…"); // \u2026 Unicode symbol, not just three separate dots
> ```

QStringLiteral

---

*[src/widget/form/groupchatform.cpp, line 211 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrufzFyGPULvk13-8tQ:-KrufzFz602XjqkV-yer:b-jop5nd) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L211)):*
> ```cpp
>     int peerNumber = 0;
>     for (const QString& fullName : names) {
>         const QString editedName = editName(fullName) + ", ";
> ```

#include <QStringBuilder>.
editName(fullName) % QLatin1String(", ");
*Especially* in for loop.
https://github.com/qTox/qTox/blob/master/doc/coding_standards.md#strings

---

*[src/widget/form/groupchatform.cpp, line 212 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrufSA2AULkFUuOpOcO:-KrufSA2AULkFUuOpOcP:b74gok3) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L212)):*
> ```cpp
>     for (const QString& fullName : names) {
>         const QString editedName = editName(fullName) + ", ";
>         QLabel* label = new QLabel(editedName);
> ```

It could be QLabel* const label, readers will see that pointer is no longer changed.

---

*[src/widget/form/groupchatform.cpp, line 218 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrugXtQ8RaXFEhPRXLE:-KrugXtQ8RaXFEhPRXLF:bxctj9u) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L218)):*
> ```cpp
>         label->setTextFormat(Qt::PlainText);
>         if (group->isSelfPeerNumber(peerNumber)) {
>             label->setStyleSheet("QLabel {color : green;}");
> ```

QStringLiteral

---

*[src/widget/form/groupchatform.cpp, line 235 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-Kruh_ubC9Oaq_LrRYKm:-Kruh_ubC9Oaq_LrRYKn:b-4nc9dy) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L235)):*
> ```cpp
>     });
>     // remove comma from last sorted label
>     QLabel* lastLabel = nickLabelList.last();
> ```

Could be QLabel* const lastLabel = ...

---

*[src/widget/form/groupchatform.cpp, line 246 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-Krugua3z66DAovruCcE:-Krugua4k4-vzmJaBklE:bxctj9u) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L246)):*
> ```cpp
> void GroupChatForm::peerAudioPlaying(int peer)
> {
>     peerLabels[peer]->setStyleSheet("QLabel {color : red;}");
> ```

QStringLiteral

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

noavarice commented:
Review status: 1 of 3 files reviewed at latest revision, 8 unresolved discussions, some commit checks pending.

---

*[src/widget/form/groupchatform.cpp, line 50 at r1](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrR1Mlxuprxydtu1rkK:-KrzCVdG6-YiiOnaayZS:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/ef6616d762d3cc353086b68436ea995bccc7594c/src/widget/form/groupchatform.cpp#L50)):*
<details><summary><i>Previously, Talkless (Vincas Dargis) wrote…</i></summary><blockquote>

It is recommended to use new QRegularExpression. See https://doc.qt.io/qt-5/qregularexpression.html#notes-for-qregexp-users .
Also, QStringLiteral("[...
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 58 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-KruhAQHimtRiRy8vA0Q:-KrzCu_lz8qkWZlPfIdP:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L58)):*
<details><summary><i>Previously, Talkless (Vincas Dargis) wrote…</i></summary><blockquote>

QStringLiteral
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 211 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrufzFyGPULvk13-8tQ:-KrzFbOy581ZV7fEYZX6:b-nvx3cq) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L211)):*
<details><summary><i>Previously, Talkless (Vincas Dargis) wrote…</i></summary><blockquote>

#include <QStringBuilder>.
editName(fullName) % QLatin1String(", ");
*Especially* in for loop.
https://github.com/qTox/qTox/blob/master/doc/coding_standards.md#strings
</blockquote></details>

I hope, `append(QLatin1String)` will do the same in more clear way. Please, let me know if it is not an equal replacement

---

*[src/widget/form/groupchatform.cpp, line 212 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrufSA2AULkFUuOpOcO:-KrzGBuler_uzMgqqMAu:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L212)):*
<details><summary><i>Previously, Talkless (Vincas Dargis) wrote…</i></summary><blockquote>

It could be QLabel* const label, readers will see that pointer is no longer changed.
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 218 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrugXtQ8RaXFEhPRXLE:-KrzGJppf4OwsRLEBgA4:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L218)):*
<details><summary><i>Previously, Talkless (Vincas Dargis) wrote…</i></summary><blockquote>

QStringLiteral
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 235 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-Kruh_ubC9Oaq_LrRYKm:-KrzGQBTHgFFaFB5P2bY:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L235)):*
<details><summary><i>Previously, Talkless (Vincas Dargis) wrote…</i></summary><blockquote>

Could be QLabel* const lastLabel = ...
</blockquote></details>

Done.

---

*[src/widget/form/groupchatform.cpp, line 246 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-Krugua3z66DAovruCcE:-KrzGljbWMHzCqMf4oED:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L246)):*
<details><summary><i>Previously, Talkless (Vincas Dargis) wrote…</i></summary><blockquote>

QStringLiteral
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

Talkless commented:
Review status: 1 of 3 files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[src/widget/form/groupchatform.cpp, line 211 at r4](https://reviewable.io:443/reviews/qtox/qtox/4573#-KrufzFyGPULvk13-8tQ:-KrzPjx6v8LJ-aiZZ84O:bcqiwkz) ([raw file](https://github.com/qtox/qtox/blob/2399a52377914d26e56eaf0dac328780b8232bcf/src/widget/form/groupchatform.cpp#L211)):*
<details><summary><i>Previously, noavarice wrote…</i></summary><blockquote>

I hope, `append(QLatin1String)` will do the same in more clear way. Please, let me know if it is not an equal replacement
</blockquote></details>

Well... I guess so. It also could be += . Although appending to a temporary feels "strange"...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4573)*
<!-- Sent from Reviewable.io -->
